lib/client.ts

```diff
@@ -61,19 +53,15 @@ const baseRequest = (defaults: RequestInit): Function => (url: string, config: R
     const request = fetch(url, opts)
 
     return request
-      .then(authValid)
-      .then((response) => new Promise((res, rej) => {
-        if (response.status === 204) {
-          return res({ body: {}, request, response });
-        }
-
+      .then((response) => {
         return response
           .json()
-          .then((body) => {
-            res({ body, request, response })
-          })
-          .catch(rej);
-      }))
+          .then((body) => new Promise((res, rej) => {
+            const sdkResp = {body, request, response}
+
+            response.ok ? res(sdkResp) : rej(sdkResp)
```

Comment:
`response.ok` returns false for 401 and 403, so those special cases aren't required.